hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestNettyIPC.java

```diff
@@ -0,0 +1,1921 @@
+/**
```

Comment:
It's impossible to review a thousand+ line test file.

Reply:
This is TestIPC with netty turned on. I noticed that parameterizing the unit tests and running both netty and non-netty was causing a lot of test failures. These failures just went away when we separated the netty and non-netty tests.

I thought as a first step I will just duplicate the test in a separate file and check whether things pass. This seems to work. A lot of the failures have gone away.

In my next PR, I will eliminate the redundancy.

Sorry for the inconvenience. But please rest assure that the test code will pretty much remain the same.

Reply:
In the latest commit I have removed the redundancy in this file.